awx/main/models/credential/__init__.py

```diff
@@ -1227,3 +1256,45 @@ def tower(cls):
             }
         },
     )
+
+
+class CredentialInputSource(PrimordialModel):
+
+    class Meta:
+        app_label = 'main'
+        unique_together = (('target_credential', 'input_field_name'),)
+
+    target_credential = models.ForeignKey(
+        'Credential',
+        related_name='input_source',
+        on_delete=models.CASCADE,
+        null=True,
+    )
+    source_credential = models.ForeignKey(
+        'Credential',
+        related_name='target_input_source',
+        on_delete=models.CASCADE,
+        null=True,
+    )
+    input_field_name = models.CharField(
+        max_length=1024,
+        default=False,
+        editable=False,
+    )
+
```

Comment:
I'm thinking we probably want some additional pre-save validation here: something that prevents a save and throws an exception if the target credential is an `external` credential or if the source credential anything but an `external` credential.